src/storage/src/client/connections.rs

```diff
@@ -713,3 +729,13 @@ impl Arbitrary for PostgresConnection {
             .boxed()
     }
 }
+
+/// A connection to a SSH tunnel.
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct SshConnection {
+    pub host: String,
+    pub port: i32,
+    pub user: String,
+    pub public_key: String,
```

Comment:
These keys could arguably be `Vec<u8>`, but it's also pretty reasonable to keep them as the human readable strings here and force the lower layers to convert those to the usable bytes.